internal/pkg/cli/svc_init.go

```diff
@@ -420,6 +432,26 @@ func parseHealthCheck(df dockerfileParser) (*manifest.ContainerHealthCheck, erro
 	}, nil
 }
 
+func (o initSvcOpts) getOSArch() (operatingSystem, architecture string, err error) {
+	var os string
+	var arch string
+	if o.image != "" {
+		os = runtime.GOOS
+		arch = runtime.GOARCH
+	} else {
+		os, arch, err = o.dockerEngineValidator.GetPlatform()
```

Comment:
What do you think of renaming from`dockerEngineValidator` to just `dockerEngine`?